File: 1878-CheckIfArrayIsSortedAndRotated/1878-CheckIfArrayIsSortedAndRotated.c
// Last updated: 6/22/2026, 12:52:11 PM
bool check(int* nums, int numsSize) {
    int firstNum = nums[0];
    int i;
    for(i=0; i<numsSize-1; i++){
        if(nums[i] > nums[i+1]) break;
    }
    for(++i; i<numsSize; i++){
        if(nums[i] > firstNum) return false;
        
        if(i == numsSize-1) continue;
        if(nums[i] > nums[i+1]) return false;
    }
    return true;
}

File: 1878-CheckIfArrayIsSortedAndRotated/1878-CheckIfArrayIsSortedAndRotated.js
// Last updated: 6/22/2026, 12:52:04 PM
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var check = function(nums) {
    let isFoundRotation = false;

    for(let i=0; i<nums.length-1; i++) {
        if(nums[i] > nums[i+1]) {
            if(isFoundRotation) return false;
            isFoundRotation = true;
        }
    }

    if(isFoundRotation) {
        return nums[0] >= nums[nums.length-1];
    }

    return nums[0] <= nums[nums.length-1];
};